icefoganalytics/sfa-client
# Enhancement: Add Spouse Section to Student Application Endpoint Response

Relates to https://github.com/icefoganalytics/student-financial-aid/issues/2
Fixes https://github.com/icefoganalytics/sfa-client/issues/41

# Context

As part of the "view submitted applications and their current status" project, this task covers returning the student spousal data.

# Data Format

```json
{
  "id": "some-application-id",
  "spouse": {
    "hasSpouse": "boolean",
    "spouse": {
      "firstName": "string",
      "lastName": "string",
      "sin": "string",
      "livingWith": "boolean",
      "postSecondaryFrom": "date",
      "postSecondaryTo": "date",
      "busService": "boolean",
      "distanceFromSchool": "number",
      "employmentInformation": "boolean",
      "employmentStatus": "string"
    }
  }
}
```

# Testing Instructions

1. Boot the `sfa-client` repo via `API_PORT=3100 dev up`
2. Boot the `student-financial-aid` front- and back-ends.
3. Log in to the app at http://localhost:8080/
4. Go to http://localhost:3000/api/portal/students/1/applications/2198 and you will see the mostly blank student spouse section. 
5. Go to http://localhost:3000/api/portal/students/8911/applications/6912 and you will see that spouse section has some content.
